l3kernel/l3prop.dtx

```diff
@@ -49,32 +49,73 @@
 %
 % \begin{documentation}
 %
-% \pkg{expl3} implements a property list data type, which contain
-% an unordered list of entries each of which consists of a \meta{key} and
-% an associated \meta{value}. The \meta{key} and \meta{value} may both
-% be any balanced text, and the \meta{key} is processed using
-% \cs{tl_to_str:n}, meaning that category codes are ignored. It is possible to
-% map functions to property lists such that the function is applied to every
-% key--value pair within the list.
+% \pkg{expl3} implements a \enquote{property list} data type, which contain
+% an unordered list of entries each of which consists of a \meta{key} (string)
+% and an associated \meta{value} (token list). The \meta{key} and \meta{value}
+% may both be given as any balanced text, and the \meta{key} is processed using
+% \cs{tl_to_str:n}, meaning that category codes are ignored. Entries can be
+% manipulated individually, as well as collectively by applying a function to
+% every key--value pair within the list.
 %
 % Each entry in a property list must have a unique \meta{key}: if an entry is
 % added to a property list which already contains the \meta{key} then the new
 % entry overwrites the existing one. The \meta{keys} are compared on a
 % string basis, using the same method as \cs{str_if_eq:nnTF}.
 %
 % Property lists are intended for storing key-based information for use within
-% code.  This is in contrast to key--value lists, which are a form of
+% code. They can be converted from and to key--value lists, which are a form of
 % \emph{input} parsed by the \pkg{l3keys} module.
 %
+% Internally, property lists can use two distinct implementations with different
+% data storage, which are decided when declaring the property list variable
+% using \cs{prop_new:N} (\enquote{flat} storage) or \cs{prop_new_large:N}
+% (\enquote{linked} storage). Once a property list is declared with
+% \cs{prop_new:N} or \cs{prop_new_large:N}, the type of internal data storage
+% can no longer be changed. All other \pkg{l3prop} functions transparently
```

Comment:
Easy to provide but (1) what name? (2) for local props it's probably annoying if people try to change the storage type within a group, I'll have to see how much bookkeeping is needed.

Reply:
I would agree that changing this “type” only locally does not sound like a good idea. 

Name: \prop_make_large:N

Reply:
Note that \prop_set_eq:NN will do the conversion if needed (both ways). 
So it just needs a small extension to cope with the name staying the same??

Reply:
`\prop_make_...:N` is a good name.  I also thought of `\prop_renew_...:N` but it might suggest that the process would give an empty prop like `\prop_new_...:N`.  Regardless of name we need both directions I suppose.

There is a bit more work than for `\prop_set_eq:NN` because some auxiliary structures have to be added / deleted.